common/flatpak-run-wayland.c

```diff
@@ -30,6 +34,38 @@
 
 #include "flatpak-utils-private.h"
 
+static char *
+get_wayland_socket_path (void)
+{
+  g_autofree char *user_runtime_dir = flatpak_get_real_xdg_runtime_dir ();
+  const char *wayland_display;
+
+  wayland_display = g_getenv ("WAYLAND_DISPLAY");
+  if (!wayland_display)
+    wayland_display = "wayland-0";
+
+  if (wayland_display[0] == '/')
+    return g_strdup (wayland_display);
+
+  return g_build_filename (user_runtime_dir, wayland_display, NULL);
+}
+
+static int
+get_wayland_socket_fd (void)
+{
+  const char *wayland_socket_fd;
+  guint64 fd;
+
+  wayland_socket_fd = g_getenv ("WAYLAND_SOCKET");
+  if (!wayland_socket_fd)
+    return -1;
+
+  if (!g_ascii_string_to_unsigned (wayland_socket_fd, 10, 0, INT_MAX, &fd, NULL))
+    return -1;
+
+  return fd;
```

Comment:
I'd write this as

```suggestion
  return (int) fd;
```

to emphasize that you're intentionally truncating it to a smaller range (which is OK to do, because `g_ascii_string_to_unsigned()` already checked that it was in the range 0 to `INT_MAX` inclusive).

Reply:
applied